src/components/implementation/no_interface/scheddev/sl.h

```diff
@@ -73,11 +73,29 @@ sl_thd_setprio(struct sl_thd *t, tcap_prio_t p)
 
 static inline struct sl_thd *
 sl_thd_lkup(thdid_t tid)
-{ return sl_mod_thd_get(sl_thd_lookup_backend(tid)); }
+{
+	assert(tid != 0);
+	if (tid > MAX_NUM_THREADS) {
```

Comment:
I'd also de-emphasize the error case (which is the prominent part of the function right now) by putting the `return NULL;` on the same line as the `if`, and not use `{}`.  This is a common pattern for error paths to de-emphasize them, and make the main path easier to read.